### ISSUE TITLE
Eye of the Beholder III problems with Sound Blaster/Adlib sound

### ISSUE THREAD

**Issue by jschwartzenberg** [dosemu2/dosemu2#1705]
**Describe the bug**
When trying to set the game to use a Sound Blaster or Aslib, it hangs on trying to detect it. Also setting it up through DOSBox and starting the game shows an error related to initializing the Sound Blaster.
![Screenshot_20220626_204637](https://user-images.githubusercontent.com/877112/175829435-760fa4b8-e069-4af6-a3f7-c654a6002c1c.png)

**To Reproduce**
Run `setup.exe` and try to set up either a Sound Blaster variant or Adlib for audio.

**Attach program/game binaries or provide an URL**
[eob3.tar.gz](https://github.com/dosemu2/dosemu2/files/8987284/eob3.tar.gz)

**Attach the log**
[boot.log](https://github.com/dosemu2/dosemu2/files/8987287/boot.log)

**A regression?**
unknown

**Comment by stsp** [comment]
Some silly timer calibration,
as usual.

**Comment by jschwartzenberg** [comment]
Seems Fantasy Empires uses similar stuff but passes:
![Screenshot_20220702_194753](https://user-images.githubusercontent.com/877112/177011124-13821265-a6b9-436c-b2c1-d254956f64fe.png)

**Comment by stsp** [comment]
Newer version...
Would one replace another?

**Comment by jschwartzenberg** [comment]
Maybe... here's a copy of the other game. I'll see if I can find time but otherwise anybody else can check both out:
[fantasy.tar.gz](https://github.com/dosemu2/dosemu2/files/9034280/fantasy.tar.gz)

**Comment by jschwartzenberg** [comment]
Another game with this sound engine:
![Screenshot_20220703_181222](https://user-images.githubusercontent.com/877112/177048089-331f19a7-039f-4c1f-a7f9-1c6dafb9f4f8.png)

This one also seems to hang.

**Comment by jschwartzenberg** [comment]
And another one, hanging too:
![Screenshot_20220917_211728](https://user-images.githubusercontent.com/877112/190873131-7d2f3506-3d34-40ba-b264-dad5c1e3abad.png)

**Comment by stsp** [comment]
I tried your "fantasy" game.
Set it to Roland for music and
Sound Blaster 16 ASP for audio
and it didn't hang.
What am I doing wrongly?

**Comment by jschwartzenberg** [comment]
Yes that one works despite that it appears to use the same sound support library as the games that fail.
